include/rocksdb/memory_allocator.h

```diff
@@ -36,9 +36,40 @@ class MemoryAllocator {
   }
 };
 
+namespace jemalloc {
+
+enum class PerCPUArena : int {
+  // Disable per-CPU arena. One single arena is used.
+  kDisabled = 0,
+
+  // Create one arena for each CPU, to reduce arena mutex contention,
+  // but may increase memory fragmentation.
+  kPerCPU = 1,
+
+  // Assume hyper-threading is enabled and create one arena for each physical
+  // CPU (half of #CPU in total). The allocator does not have run-time check to 
+  // verify whether the assumption is true. This mode may have less memory
+  // fragmentation comparing to kPerCPU, for using less arenas.
+  kPerPhysicalCPU = 2,
+};
+
+struct JemallocAllocatorOptions {
```

Comment:
I didn't name it `JemallocNodumpAllocatorOptions` because I plan to add a generic `JemallocAllocator`. The generic interface can be used by users to implement custom allocator similar to the no dump allocator.